06_puppy/sirigithub/stores_test.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store     Storer
+	newStorer func() Storer
+	puppy1    Puppy
+}
+
+func (suite *storesSuite) SetupTest() {
+	suite.store = suite.newStorer()
+	suite.puppy1 = Puppy{Breed: "cavalier", Color: "White", Value: "$1700"}
+	suite.store.CreatePuppy(&suite.puppy1)
+}
+func (suite *storesSuite) TestCreatePuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	newPuppy := Puppy{Breed: "Dobberman", Color: "Black", Value: "$800"}
+	//when
+	id := suite.store.CreatePuppy(&newPuppy)
+	//then
+	assert.True(newPuppy.ID == suite.puppy1.ID+1, "Puppy Id increments sequentially")
```

Comment:
This is an equality test, so you should use `assert.Equal(suite.puppy1.ID+1, newPuppy,ID, "...")`